src/traces/sankey/defaults.js

```diff
@@ -0,0 +1,136 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+var d3 = require('d3');
+
+
+function linksDefaults(traceIn, traceOut) {
+    var linksIn = traceIn.links || [],
+        linksOut = traceOut.links = [];
+
+    var linkIn, linkOut, i;
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(linkIn, linkOut, attributes.links, attr, dflt);
+    }
+
+    for(i = 0; i < linksIn.length; i++) {
+        linkIn = linksIn[i];
+        linkOut = {};
+
+        if(!Lib.isPlainObject(linkIn)) {
+            continue;
+        }
+
+        var visible = coerce('visible');
+
+        if(visible) {
+            coerce('label');
+            coerce('value');
+            coerce('source');
+            coerce('target');
+            coerce('color');
+        }
+
+        linkOut._index = i;
+        linksOut.push(linkOut);
+    }
+}
+
+function nodesDefaults(traceIn, traceOut) {
+    var nodesIn = traceIn.nodes || [],
+        nodesOut = traceOut.nodes = [];
+
+    var nodeIn, nodeOut, i, j, link, foundUse, visible,
+        usedNodeCount = 0,
+        indexMap = [];
+
+    var defaultPalette = d3.scale.category20();
```

Comment:
I guess our 10-color default [palette](https://github.com/plotly/plotly.js/blob/b32c21d9af65149e4ee7987ae66336191707eee0/src/components/color/attributes.js#L13) isn't ideal for Sankey. I would still for using it for Sankey for consistency.

Thoughts?